src/xdist/remote.py

```diff
@@ -155,6 +155,8 @@ def pytest_runtestloop(self, session):
         self.nextitem_index = self._get_next_item_index()
         while self.nextitem_index is not self.SHUTDOWN_MARK:
             self.run_one_test()
+            if session.shouldfail or session.shouldstop:
+                break
```

Comment:
There might be a better way to do this - perhaps testing `self.torun` for _any_ `self.SHUTDOWN_MARK` items, even if those items aren't the "next" ones in the queue? I can't find much API info on what data type `torun` actually is though.

Reply:
This change seems reasonable enough to me. 👍